spec/index.html

```diff
@@ -5669,248 +5670,253 @@ <h5>NOT IN</h5>
-_:c3  :weight       "85"^^t:kilos .
-_:c3  :displacement  "85"^^t:liters .
-              </pre>
-              <div class="queryGroup">
-                <pre class="query nohighlight">
-PREFIX  :      &lt;http://example.org/WMterms#&gt;
-PREFIX  t:     &lt;http://example.org/types#&gt;
+            <div class="ednote" style="background-color: #EEE" >
+              <p>If the function is to become callable as 
+                <em>`sameValue(term1, term2)`</em>, add a rule 
+                that if there is an
+                <a href="#OperatorMapping">operation mapping</a> for "="
+                for the arguments, an implementation MUST return the same boolean value
+                from <em>`sameValue(term1, term2)`</em>.
+              </p>
+            </div>
+            <div class="ednote" style="background-color: #EEE" >
+              <p>
+                The treatment of `NaN` can not be consistent with both "same term means same value" 
+                and "`fn:numeric-equal(NaN, NaN)` is false". So it is an arbitrary choice with
+                arguments for all three possible choices.
+                For pattern matching, `"NaN"^^xsd:double` must match itself and
+                `"NaN"^^xsd:double` does not pattern match `"NaN"^^xsd:float`.
+              </p>
+              <p>
+                The operator mapping for `=` goes to `fn:numeric-compare` which is false.
+                The `sameValue` result could be made an error (or false) as a special case.
+              </p>
+              <p>
+                `"NaN"^^xsd:double = "NaN"^^xsd:float` is also handled by the
+                operator mapping and is false &ndash; `xsd:float` is promoted to `xsd:double`.
+                A reading of IEEE 754 is that `"NaN"^^xsd:double` and `"NaN"^^xsd:float`
+                are the same term also suggests that `sameValue` is true.
+              </p>
+              <p>
+                <b>Proposal 1</b>: follow "sameTerm means sameValue" and `sameValue("NaN"^^xsd:double, "NaN"^^xsd:double)`
+                and `sameValue("NaN"^^xsd:float, "NaN"^^xsd:float)` are <i>true</i>s.
+                <br/>
+                <b>Proposal 2</b>: 
+                follow "there is one NaN" so `sameValue("NaN"^^xsd:double, "NaN"^^xsd:float)` is <i>true</i>.
+              </p>
+              <p>
+                Ref</i>: <a href="https://www.w3.org/TR/xmlschema11-2/#dt-specialvalue">XSD schema 1.1 special values</a>
+                "are members of the datatype's ·value space·. ". It is not clear whether "special value" literals
+                added to the value spaces of `xsd:double` and `xsd:float` are "the same" literals.
+              <p>
+                <i>Ref</i>: <a href="https://en.wikipedia.org/wiki/IEEE_754#Special_values">(wikipedia) IEEE_754 Special Values</a>
+                <i>IEEE 754 specifies a special value called "Not a Number" (NaN)</i>.
+                suggests they are the same.
```

Comment:
Per this, I think proposal 2 is definitely better and less surprising than proposal 1

Reply:
> I don't think @afs meant these two proposals to exclude one another.

Indeed, my bad. Thanks!

> the problem is not actually relevant for the definition of `sameValue`

I am not fully convinced of this because it means that because `xsd:float` has an entry in the operator mapping (`fn:numeric-compare`) then it is allowed to defined that `"NaN"^^xsd:float = "NaN"^^xsd:float` is FALSE whereas a custom `ex:myFloat` won't be able to do that because of the `sameValue` definition that mandates that `sameTerm` implies `sameValue`. This introduces a discrepency between built-in datatypes and the others and prevent to consider `sameValue` definition as working for all terms including `xsd:float` literals.

Reply:
To add to the fun, we also have `0` and `-0`

xsd:double:
> Equality is identity, except that  0 = −0  (although they are not identical) and  NaN ≠ NaN  (although NaN is of course identical to itself).

Reply:
As per @tpt's point about extension, lets have one "NaN" value and it is the same in xsd:doube and xsd:float value spaces.

`sameValue("NaN"^^xsd:double, "NaN"^^xsd:float)` is true.